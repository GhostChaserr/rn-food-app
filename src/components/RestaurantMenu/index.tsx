import React, { useState, FC } from 'react'
import { ImageSourcePropType } from 'react-native'

import {
  RestaurantMenuWrapper,
  StyledMenuImage,
  StyledQuantityBtnsContainer,
  StyledQuantityBtn,
  StyledQuantityBtnText,
  StyledMenuImageWrapper,
  StyledMenuContentWrapper,
  StyledMenuNameContainer,
  StyledMenuName,
  StyledMenuDescriptionContainer,
  StyledMenuDescription
} from './styles'

type RestaurantMenuProps = {
  photo: ImageSourcePropType
  name: string
  description: string
  menuId: number
}

const RestaurantMenu: FC<RestaurantMenuProps> = (props) => {
  const [quantity, setQuantity] = useState<number>(0)

  const handleQuantityIncrement = () => setQuantity(quantity + 1)
  const handleQuantityDecrement = () => setQuantity(quantity - 1)
  return (
    <RestaurantMenuWrapper>
      <StyledMenuImageWrapper>
        <StyledMenuImage resizeMode="cover" source={props.photo} />
      </StyledMenuImageWrapper>
      <StyledMenuContentWrapper>
        <StyledMenuNameContainer>
          <StyledMenuName>{props.name}</StyledMenuName>
        </StyledMenuNameContainer>
        <StyledMenuDescriptionContainer>
          <StyledMenuDescription>{props.description}</StyledMenuDescription>
        </StyledMenuDescriptionContainer>
      </StyledMenuContentWrapper>
      <StyledQuantityBtnsContainer>
        <StyledQuantityBtn onPress={handleQuantityDecrement} position="left">
          <StyledQuantityBtnText> - </StyledQuantityBtnText>
        </StyledQuantityBtn>
        <StyledQuantityBtn disabled={true} position="center">
          <StyledQuantityBtnText> {quantity} </StyledQuantityBtnText>
        </StyledQuantityBtn>
        <StyledQuantityBtn onPress={handleQuantityIncrement} position="right">
          <StyledQuantityBtnText> + </StyledQuantityBtnText>
        </StyledQuantityBtn>
      </StyledQuantityBtnsContainer>
    </RestaurantMenuWrapper>
  )
}

export default RestaurantMenu
