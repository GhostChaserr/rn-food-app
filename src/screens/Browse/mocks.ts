const categoriesMocks = [
  {
    id: 'plants',
    name: 'Plants',
    tags: ['products', 'inspirations'],
    count: 147,
    image: require('src/assets/icons/plants.png')
  },
  {
    id: 'seeds',
    name: 'Seeds',
    tags: ['products', 'shop'],
    count: 16,
    image: require('src/assets/icons/seeds.png')
  },
  {
    id: 'flowers',
    name: 'Flowers',
    tags: ['products', 'inspirations'],
    count: 68,
    image: require('src/assets/icons/flowers.png')
  },
  {
    id: 'sprayers',
    name: 'Sprayers',
    tags: ['products', 'shop'],
    count: 17,
    image: require('src/assets/icons/sprayers.png')
  },
  {
    id: 'pots',
    name: 'Pots',
    tags: ['products', 'shop'],
    count: 47,
    image: require('src/assets/icons/pots.png')
  },
  {
    id: 'fertilizers',
    name: 'fertilizers',
    tags: ['products', 'shop'],
    count: 47,
    image: require('src/assets/icons/fertilizers.png')
  }
]

export { categoriesMocks }
