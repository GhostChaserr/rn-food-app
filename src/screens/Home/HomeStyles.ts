import styled from 'styled-components/native'

export const StyledSafeAreaView = styled.SafeAreaView`
  flex: 1;
`
export const StyledContentWrapper = styled.View`
  flex: 1;
  padding: 20px;
`

export const StyledHeaderWrapper = styled.View`
  justify-content: space-between;
  flex-direction: row;
`

export const StyledTouchableIcon = styled.TouchableOpacity``

export const StyledHeaderIcon = styled.Image`
  width: 20px;
  height: 20px;
  position: relative;
  top: 5px;
`

export const StyledHeaderLocation = styled.View`
  justify-content: center;
  background-color: rgba(0,0,0, 0.1);
  padding: 10px 60px;
  border-radius: 10px;
`

export const StyledHeaderLocationText = styled.Text``

export const StyledHeadingsWrapper = styled.View`
  margin-top: 20px;
  flex-direction: column;
  align-items: flex-start;
`

export const StyledHeadingsHeader = styled.Text`
  font-weight: bold;
  font-size: 36px;
`

export const StyledCategoriesContainer= styled.View`
  margin-top: 10px;
`

export const StyledCategoryButton = styled.TouchableOpacity<{ isSelected: boolean }>`
  padding: 10px;
  padding-bottom: 20px;
  background-color: ${({ isSelected }) => isSelected ? '#FC6D3F;': 'white;' } 
  border-radius: 30px;
  align-items: center;
  justify-content: center;
  margin-right: 10px;
`

export const StyledCategoryWrapper = styled.View`
  width: 50px;
  height: 50px;
  align-items: center;
  justify-content: center;
  background-color: white;
  border-radius: 30px;
`

export const StyledCategoryIcon = styled.Image`
  resize-mode: contain;
  width: 35px;
  height: 35px;
`

export const StyledCategoryTextWrapper = styled.View`
  margin-top: 10px;
  align-items: center;
  justify-content: center;
`

export const StyledFlatList = styled.FlatList`
 
`

export const StyledCategoryText = styled.Text<{ isSelected: boolean }>`
  font-size: 11px;
  color: ${({ isSelected }) => isSelected ? 'white;': 'black;' } 
  font-weight: bold;
`

export const StyledRestaurantsFeedWrapper = styled.View`
  margin-top: 10px;
`

