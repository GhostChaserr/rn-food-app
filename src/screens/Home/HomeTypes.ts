import { StackNavigationProp } from '@react-navigation/stack'
import { createStackNavigator } from '@react-navigation/stack'

type RootStackParamList = {
  Home: undefined;
  Profile: { userId: string };
  Feed: { sort: 'latest' | 'top' } | undefined;
};

const RootStack = createStackNavigator<RootStackParamList>();

type ProfileScreenNavigationProp = StackNavigationProp<
  RootStackParamList,
  'Profile'
>;


export type RestaurantCategory = {
  id: number
  icon: any
  name: string
}