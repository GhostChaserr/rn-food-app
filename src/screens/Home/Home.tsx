import React, { useState } from 'react'
import GlobalStyles from 'src/global.styles'
import { icons } from 'src/constants'

import Restaurant from 'src/components/Restaurant'
import { categoryData, restaurantData } from './mocks'

import {
  StyledSafeAreaView,
  StyledHeaderWrapper,
  StyledTouchableIcon,
  StyledHeaderIcon,
  StyledHeaderLocation,
  StyledHeaderLocationText,
  StyledContentWrapper,
  StyledHeadingsWrapper,
  StyledHeadingsHeader,
  StyledCategoriesContainer,
  StyledCategoryButton,
  StyledCategoryWrapper,
  StyledCategoryIcon,
  StyledCategoryTextWrapper,
  StyledCategoryText,
  StyledFlatList,
  StyledRestaurantsFeedWrapper
} from './HomeStyles'

const Home = () => {
  const [restaurants, setRestaurants] = useState<any>(restaurantData)
  const [categories, setCategoris] = useState(categoryData)
  const [selectedCategory, setSelectedCategory] = React.useState<any>(null)

  const onCategorySelect = (category: any) => setSelectedCategory(category)

  const renderCategory = ({ item }: any) => {
    const isSelected = selectedCategory?.id == item.id
    return (
      <StyledCategoryButton
        isSelected={isSelected}
        onPress={() => onCategorySelect(item)}
      >
        <StyledCategoryWrapper>
          <StyledCategoryIcon source={item.icon} />
        </StyledCategoryWrapper>
        <StyledCategoryTextWrapper>
          <StyledCategoryText isSelected={isSelected}>
            {item.name}
          </StyledCategoryText>
        </StyledCategoryTextWrapper>
      </StyledCategoryButton>
    )
  }

  const renderRestaurant = ({ item }: any) => {
    return <Restaurant  {...item} />
  }

  return (
    <StyledSafeAreaView style={GlobalStyles.adroidSafeArea}>
      <StyledContentWrapper>
        <StyledHeaderWrapper>
          <StyledTouchableIcon>
            <StyledHeaderIcon source={icons.nearby} />
          </StyledTouchableIcon>
          <StyledHeaderLocation>
            <StyledHeaderLocationText>
              {' '}
              45 Lincoln street{' '}
            </StyledHeaderLocationText>
          </StyledHeaderLocation>
          <StyledTouchableIcon>
            <StyledHeaderIcon source={icons.basket} />
          </StyledTouchableIcon>
        </StyledHeaderWrapper>

        <StyledHeadingsWrapper>
          <StyledHeadingsHeader>Main</StyledHeadingsHeader>
          <StyledHeadingsHeader>Categoris</StyledHeadingsHeader>
        </StyledHeadingsWrapper>

        <StyledCategoriesContainer>
          <StyledFlatList
            data={categories}
            horizontal
            showsHorizontalScrollIndicator={false}
            keyExtractor={(item: any) => `${item.id}`}
            renderItem={renderCategory}
          />
        </StyledCategoriesContainer>
        <StyledRestaurantsFeedWrapper>
          <StyledFlatList
            data={restaurantData}
            keyExtractor={(item: any) => `${item.id}`}
            renderItem={renderRestaurant}
          />
        </StyledRestaurantsFeedWrapper>
      </StyledContentWrapper>
    </StyledSafeAreaView>
  )
}

export default Home
