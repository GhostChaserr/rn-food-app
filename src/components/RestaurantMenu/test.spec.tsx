import React from 'react'
import TestRenderer from 'react-test-renderer'
import { restaurantData } from 'src/screens/Home/mocks'
import RestaurantMenu from './index'

describe('Testing Menu Component', () => {
  it('Should render Restaurant Menu Component', () => {
    const menu = restaurantData[0].menu[0];
    const tree = TestRenderer.create(
      <RestaurantMenu
        name={menu.name}
        photo={menu.photo}
        menuId={menu.menuId}
        description={menu.description}
      />
    )
    expect(tree).toMatchSnapshot()
  })
})
