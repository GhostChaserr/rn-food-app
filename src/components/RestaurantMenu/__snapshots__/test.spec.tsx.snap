// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Menu Component Should render Restaurant Menu Component 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "width": 750,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <Image
      resizeMode="cover"
      source={1}
      style={
        Array [
          Object {
            "height": 350,
            "width": "100%",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
          "justifyContent": "center",
          "marginTop": 30,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 24,
              "fontWeight": "bold",
              "textAlign": "center",
            },
          ]
        }
      >
        Crispy Chicken Burger
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "textAlign": "center",
            },
          ]
        }
      >
        Burger with crispy chicken, cheese and lettuce
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "bottom": 110,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "center",
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomLeftRadius": 25,
          "borderBottomRightRadius": 0,
          "borderTopLeftRadius": 25,
          "borderTopRightRadius": 0,
          "justifyContent": "center",
          "opacity": 1,
          "position": "relative",
          "right": 10,
          "width": 50,
        }
      }
    >
      <Text
        style={
          Array [
            Object {},
          ]
        }
      >
         - 
      </Text>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomLeftRadius": 0,
          "borderBottomRightRadius": 0,
          "borderTopLeftRadius": 0,
          "borderTopRightRadius": 0,
          "justifyContent": "center",
          "opacity": 1,
          "position": "relative",
          "right": 10,
          "width": 50,
        }
      }
    >
      <Text
        style={
          Array [
            Object {},
          ]
        }
      >
         
        0
         
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomLeftRadius": 0,
          "borderBottomRightRadius": 25,
          "borderTopLeftRadius": 0,
          "borderTopRightRadius": 25,
          "justifyContent": "center",
          "opacity": 1,
          "position": "relative",
          "right": 10,
          "width": 50,
        }
      }
    >
      <Text
        style={
          Array [
            Object {},
          ]
        }
      >
         + 
      </Text>
    </View>
  </View>
</View>
`;
